File: core/client/game/src/commonGame/logic/func/ILogicEntity.cs
using System;
using ShineEngine;

/// <summary>
/// 逻辑主体
/// </summary>
public interface ILogicEntity:ITimeEntity
{
	/** 随机一个整形 */
	int randomInt(int range);

	/** 判定几率 */
	bool randomProb(int prob,int max);

	/** 随一整形(start<=value<end) */
	int randomRange(int start,int end);

	/** 随一整形(start<=value<=end)(包括结尾) */
	int randomRange2(int start,int end);
}

File: core/clientLaya/game/src/commonGame/logic/ILogicEntity.ts
namespace Shine
{
	/** 逻辑主体 */
	export interface ILogicEntity extends ITimeEntity
	{
		/** 随机一个整形 */
		 randomInt(rangeL:number):number;
	
		/** 判定几率 */
		 randomProb(prob:number,max:number):boolean;
	
		/** 随一整形(start<=value<end) */
		randomRange(start:number,end:number):number;
	
		/** 随一整形(start<=value<=end)(包括结尾) */
		randomRange2(start:number,end:number):number;
	}
}